### PR TITLE
feat: Join miner and repair tool into single CLI

### LINKED ISSUE #148: Add documentation for the Sonar warnings miner.

### LINKED ISSUE #223: Join miner and repair tool into single CLI
As mentioned in #221 , they should be available as subcommands of the core CLI. So, for example:

```
$ java -jar sorald.jar repair <arguments...>
$ java -jar sorald.jar mine <arguments...>
```

### PR DESCRIPTION
**Pull request by slarse**
Fix #223 
Fix #148 

Replaces #224 as Codecov gets really confused when you force-push

This PR joins the miner and repair tool into a single CLI. The top level now looks like so:

```
$ java -jar sorald.jar --help
Usage: sorald [-hV] <COMMAND>
The Sorald command line application for automatic repair of Sonar rule
violations.
  -h, --help      Show this help message and exit.
  -V, --version   Print version information and exit.
Commands:
  repair  Repair Sonar rule violations in a targeted project.
  mine    Mine a project for Sonar warnings.
```

And you for example invoke the commands like so:

```
$ java -jar sorald.jar repair --ruleKeys 2755 --originalFilesPath path/to/project
$ java -jar sorald.jar mine --originalFilesPath path/to/project
```

I found that this was also the time to address the lack of documentation for the warnings miner (see #148), which now has documented usage in the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,21 +8,55 @@ It can currently repair violations of [15+ rules](/docs/HANDLED_RULES.md).
...
-$ java -jar target/sorald-1.1-SNAPSHOT-jar-with-dependencies.jar <arguments>
  ```
 
-The arguments are the following:
+The Sorald application can now be found in
+`target/sorald-1.1-SNAPSHOT-jar-with-dependencies.jar`.
+
+### Usage
+
+Sorald can do two things: automatically repair violations of Sonar rules in a
```
Comment (fermadeiral) [inline README.md:27]: "Sorald can do two things" -> "Sorald can do two tasks"

Comment (slarse) [reply README.md:27]: How about this ?

```suggestion
Sorald can perform two different tasks: automatically repair violations of Sonar rules in a
```

Comment (fermadeiral) [reply README.md:27]: Perfect!

Comment (slarse) [reply README.md]: Done!

**Comment by fermadeiral** [comment]
This is a great PR, which improves the usability of Sorald a lot. Thank you for this, @slarse!

**Pull request merged**